analysis/topology/continuity.lean

```diff
@@ -706,14 +706,20 @@ instance [second_countable_topology α] [second_countable_topology β] :
 
 end prod
 
+section compact_and_proper_spaces
 
-section locally_compact
+/--Type class for compact spaces. Separation is sometimes included in the definition, especially
+in the French literature, but we do not include it here.-/
+class compact_space (α : Type*) [topological_space α] : Prop :=
```

Comment:
Perhaps it would be better to move this much earlier, to right after the definition of compact sets, so that we can provide all the instances like "a quotient of a compact space is compact" along the way? Or should we just write all those instances all right here?

Reply:
I agree it would make sense to have all these instances earlier in the theory: it is weird to have them in a file `continuity.lean`. But there were already many instances here, so I thought I would not refactor anything in this PR, only add some new material, and that refactoring would be for a later PR. Combining refactoring and new material makes for a much harder PR review, I think.